packages/components/src/Chips/InternalChipDismissible/__tests__/KeyboardNavigation.test.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useState } from "react";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { InternalChipDismissible } from "..";
+import { Chip } from "../..";
+
+// Helper function to get a chip by label
+function getByChipLabelText(chipName: string) {
+  return screen.getByLabelText(
+    chipName + ". Press delete or backspace to remove " + chipName,
+  );
+}
+
+// Helper to get remove button within a chip
+function getRemoveButton(chip: HTMLElement) {
+  const button = chip.querySelector('[data-testid="remove-chip-button"]');
```

Comment:
while this is just a test and I'm more open to querying in a test than I am in the actual component, we should see if we can't find a better way to find it.

actually, if it has a test-id why are we not using testing-library selectors?